Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -672,8 +696,78 @@ - (NSInteger)readIntegerFromDefaults:(NSString *)key {
     return NSNotFound;
 }
 
-- (NSString *)prefsFile {
-    return [[NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject] stringByAppendingPathComponent:BRANCH_PREFS_FILE];
-}
++ (NSString *)prefsFile_deprecated
+    {
+    NSString * path =
+        [[NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES)
+            firstObject]
+                stringByAppendingPathComponent:BRANCH_PREFS_FILE];
+    return path;
+    }
+
++ (NSURL*) URLForBranchDirectory
+    {
+    NSError *error = nil;
+    NSURL *URL =
+        [[NSFileManager defaultManager]
+            URLForDirectory:NSApplicationSupportDirectory
+            inDomain:NSUserDomainMask
+            appropriateForURL:nil
+            create:YES
+            error:&error];
+    if (error)
+        {
+        NSLog(@"Error creating URLForPrefsDirectory: %@.", error);
+        return nil;
+        }
+    URL = [URL URLByAppendingPathComponent:@"io.branch"];
+    [[NSFileManager defaultManager]
+        createDirectoryAtURL:URL
+        withIntermediateDirectories:YES
+        attributes:nil
+        error:&error];
+    if (error)
+        {
+        NSLog(@"Error creating URLForPrefsDirectory: %@.", error);
+        return nil;
+        }
+    return URL;
+    }
+
++ (NSURL*) URLForPrefsFile
+    {
+    NSURL *URL = [self URLForBranchDirectory];
+    URL = [URL URLByAppendingPathComponent:BRANCH_PREFS_FILE];
+    return URL;
+    }
+
++ (void) moveOldPrefsFile
```

Comment:
Since this function is pretty much duplicated below, maybe we can pull it out to an abstract file mover function with input/output?